test/framework/test_framework.py

```diff
@@ -340,3 +340,32 @@ def circuits(self):
         }
 
         self.assertEqual(exp.job_info(backend=backend), job_info)
+
+    @ddt.data((None, "Test1"), (None, "Test2"))
+    @ddt.unpack
+    def test_experiment_type(self, exp_type1, exp_type2):
+        """Test the experiment_type setter for the experiment."""
+
+        class MyExp(BaseExperiment):
+            """Some arbitrary experiment"""
+
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+            def circuits(self):
+                pass
+
+        exp1 = MyExp(physical_qubits=[0], experiment_type=exp_type1)
```

Comment:
Instead of passing `exp_type` to the constructor, pass some arbitrary string (like `blaaa`) and verify that the setter overrides it.